### PR TITLE
Decouple ivy component from publication model

### PR DESCRIPTION
**Pull request by jvandort**
Physically separate the logic which maps a SoftwareComponentInternal to a published ivy descriptor model This will allow us to more easily adapt publishing to new component models

A similar change was already done for maven publishing in #25129

### CODE REVIEW

File: subprojects/ivy/src/main/java/org/gradle/api/publish/ivy/internal/publication/DefaultIvyPublication.java [hunk 1]
```diff
@@ -256,192 +265,294 @@ public void setAlias(boolean alias) {
...
     private void populateFromComponent() {
         if (populated) {
             return;
         }
         populated = true;
-        if (component == null) {
-            return;
+        if (!artifactsOverridden && parsedComponent.isPresent()) {
+            mainArtifacts.addAll(parsedComponent.get().getArtifacts());
+            configurations.addAll(parsedComponent.get().getConfigurations());
```
Comment (jvandort) [inline subprojects/ivy/src/main/java/org/gradle/api/publish/ivy/internal/publication/DefaultIvyPublication.java:287]: This should be in another if statement which is not guarded by `!artifactsOverridden`

**Comment by jvandort** [comment]
@bot-gradle test this

**Review by tresat — APPROVED** [review]
So the descriptor already existed for Ivy, we just weren't using it.  This forces us to use it and decouple.  LGTM.

**Comment by jvandort** [comment]
@bot-gradle test and merge

**Comment by jvandort** [comment]
@bot-gradle test and merge

**Pull request merged**